testing/e2e/workload_test.go

```diff
@@ -421,6 +426,9 @@ func TestCreateFromGitWithAnnotations(t *testing.T) {
 				if regexProgress.FindString(output) != "" {
 					t.Fatalf("expected progressbar not to be in output %v", output)
 				}
+				if strings.Contains(output, apis.LocalSourceProxyAnnotationName) {
+					t.Fatalf("local source proxy annotation %q not expected in output %v", apis.LocalSourceProxyAnnotationName, output)
```

Comment:
Can we also add a test for failure scenarios , when the workload was created using `--source-image` and trying to update without a repo details?

Reply:
@atmandhol if user changes something in the local source and then want to apply but they do not use the source image, should this fail?

Reply:
No, you only have to provide --source-image flag once. For all subsequent calls, the source-image value is taken from the workload.spec.source.image field

Reply:
updated to reflect this behavior